### PR TITLE
LPS-163525 | Add tests HasAnUnderstandableErrorMessageWhenURLInvalid and HasMandatoryFieldErrorMessagesWhenEmpty

### PR DESCRIPTION
**Pull request by bronsonsoda**
**Test plan**
_HasAnUnderstandableErrorMessageWhenURLInvalid_
Given Add IFrame Remote Apps
When type testurl in URL field
And When a valid name is entered
And When attempt to save the remote app
Then error message "Please enter a valid URL" appears at the top of the page

_HasMandatoryFieldErrorMessagesWhenEmpty_
Given Add IFrame Remote Apps admin page
When required fields Name and URL are blank
And When attempt to save remote app
Then error message "field is required" appears under URL
And Then Remote App is not added

https://issues.liferay.com/browse/LPS-163527

### CODE REVIEW

**Comment by bronsonsoda** [comment]
ci:test:frontend:nocompile

**Comment by bronsonsoda** [comment]
I'm not sure these will pass until [the draft](https://github.com/liferay-frontend/liferay-portal/pull/2594) is merged

File: modules/apps/frontend-js/frontend-js-remote-applications/tests/RemoteApplications.testcase [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		User.firstLoginPG();
+	}
+
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		PortalInstances.tearDownCP();
+	}
+
+	@description = "LPS-163525. Assert error messege appears when invaild url is entered."
```
Comment (bronsonsoda) [inline modules/apps/frontend-js/frontend-js-remote-applications/tests/RemoteApplications.testcase:22]: asser

File: modules/apps/frontend-js/frontend-js-remote-applications/tests/RemoteApplications.testcase [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+		}
+
+		task ("Then error message 'Please enter a valid URL' appears at the top of the paged") {
+			AssertTextEquals.assertPartialText(
+				locator1 = "Message#ERROR_ENTER_A_VALID_URL",
+				value1 = "Please enter a valid URL.");
+		}
+	}
+
+	@description = "LPS-163525. Assert iFrame is not add and warnings appear."
```
Comment (bronsonsoda) [inline modules/apps/frontend-js/frontend-js-remote-applications/tests/RemoteApplications.testcase:51]: assert

File: modules/apps/frontend-js/frontend-js-remote-applications/tests/RemoteApplications.testcase [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+			RemoteApps.addApp(
+				entryName = "Test Name",
+				entryURL = "$testurl");
+		}
+
+		task ("And When attempt to save the remote app") {
+			Button.clickPublish();
+		}
+
+		task ("Then error message 'Please enter a valid URL' appears at the top of the paged") {
```
Comment (bronsonsoda) [inline modules/apps/frontend-js/frontend-js-remote-applications/tests/RemoteApplications.testcase:44]: remove top of page

File: modules/apps/frontend-js/frontend-js-remote-applications/tests/RemoteApplications.testcase [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+				portlet = "Remote Apps");
+
+			RemoteApps.addType(type = "IFrame");
+		}
+
+		task ("When required fields Name and URL are blank and When attempt to save remote app") {
+			Button.clickPublish();
+		}
+
+		task ("Then an error message 'field is required' appears under URL and Remote App is not added") {
```
Comment (john-co) [inline modules/apps/frontend-js/frontend-js-remote-applications/tests/RemoteApplications.testcase:67]: split into two Then tasks

Then error message "field is required" appears under URL
And Then Remote App is not added

**Review by john-co — APPROVED** [review]
Please make adjustments and move to RemoteApps.testcase in portalweb.

File: modules/apps/frontend-js/frontend-js-remote-applications/tests/RemoteApplications.testcase (hunk 1 above)
Comment (john-co) [reply modules/apps/frontend-js/frontend-js-remote-applications/tests/RemoteApplications.testcase:22]: "Validate Remote Apps UI has an understandable error message when user inputs an invalid URL."

File: modules/apps/frontend-js/frontend-js-remote-applications/tests/RemoteApplications.testcase (hunk 2 above)
Comment (john-co) [reply modules/apps/frontend-js/frontend-js-remote-applications/tests/RemoteApplications.testcase:51]: "Validate Remote apps UI has a clear error message when all the mandatory fields are empty."

"Test that Remote Apps has mandatory field error messages when empty."

**Comment by bronsonsoda** [comment]
ci:test:frontend:nocompile

**Comment by bronsonsoda** [comment]
ci:forward

**Pull request closed without merging**